### PR TITLE
Add support for creating private passports

### PR DESCRIPTION
**Pull request by ssaitejar96**
Will push UI side changes once this PR is merged.

### CODE REVIEW

File: packages/contracts/contracts/Passport.sol [hunk 1]
```diff
@@ -38,14 +42,16 @@ contract Passport is ERC721, Ownable {
     royaltyPcnt = _royaltyPcnt;
     royaltyRecipient = _owner;
     claimable = _claimable;
+    isPrivate = _isPrivate;
+    _setupRole(DEFAULT_ADMIN_ROLE, owner_);
```
Comment (ssaitejar96) [inline packages/contracts/contracts/Passport.sol:46]: should we also add the owner to the list of minters(MINTER_ROLE) by default as part of constructor?
I wasn't sure if we can make that assumption.

File: packages/contracts/contracts/Passport.sol [hunk 2]
```diff
@@ -118,4 +125,46 @@ contract Passport is ERC721, Ownable {
...
+      returns(bool)
+  {
+      return hasRole(MINTER_ROLE, account);
+  }
+
+  /**
+    * @dev Add accounts to the minter role. Restricted to admins.
+    * @param accounts The members to add as a member.
+    */
+  function addMinters(address[] memory accounts)
```
Comment (ssaitejar96) [inline packages/contracts/contracts/Passport.sol:147]: I made both addMinters and removeMinters accept a list of address to prepare for the CSV bulk upload

File: packages/contracts/hardhat.config.ts [hunk 3]
```diff
@@ -81,4 +81,16 @@ const config: HardhatUserConfig = {
   },
 };
 
+module.exports = {
```
Comment (ssaitejar96) [inline packages/contracts/hardhat.config.ts:84]: I do not know the side effects of this. Without this I was getting the below error:
Screenshot 2022-02-05 at 2 06 24 AM

Any idea?

File: packages/contracts/contracts/Passport.sol (hunk 1 above)
Comment (dvargasfuertes) [reply packages/contracts/contracts/Passport.sol:46]: I'd leave off for now (less gas) and see what product feedback says

File: packages/contracts/hardhat.config.ts (hunk 3 above)
Comment (dvargasfuertes) [reply packages/contracts/hardhat.config.ts:84]: I too am unsure. I wonder if we should start cutting down on the logic in `Passport.sol`. But it's possible that the optimizer just removes unused methods and fields from the dependencies. Let's see if this successfully deploys to the test networks.

**Review by dvargasfuertes — APPROVED** [review]

**Pull request merged**